tests/pyroutils_test.py

```diff
@@ -25,3 +71,133 @@ def prior(num_samples=1):
+        torch.Size([dim]),
+        torch.Size([dim]),
+    )
+    assert estimator_dist.batch_shape == torch.Size([num_simulations])
+    assert estimator_dist.event_shape == torch.Size([dim])
+    assert estimator_dist.support == torch.distributions.constraints.real
+
+    # Test sample method
+    if trainer_cls is not NRE:
+        assert estimator_dist.sample().shape == torch.Size([num_simulations, dim])
+        x_samples = estimator_dist.sample(torch.Size([3]))
+        assert x_samples.shape == torch.Size([3, num_simulations, dim])
+    else:
+        with pytest.raises(NotImplementedError):
+            estimator_dist.sample()
+        with pytest.raises(NotImplementedError):
+            estimator_dist.sample(torch.Size([3]))
+        x_samples = torch.distributions.Normal(theta, 1.0).sample(torch.Size([3]))
+
+    # Test log_prob method
+    assert estimator_dist.log_prob(x).shape == torch.Size([num_simulations])
+    assert estimator_dist.log_prob(x_samples).shape == torch.Size([3, num_simulations])
+
+    # Test expand method
+    estimator_dist_expanded = estimator_dist.expand(torch.Size([2, num_simulations]))
+    assert isinstance(estimator_dist_expanded, distribution_cls)
+    assert estimator_dist_expanded.batch_shape == torch.Size([2, num_simulations])
+    assert estimator_dist_expanded.event_shape == estimator_dist.event_shape
+    assert torch.equal(
+        estimator_dist_expanded.condition,
+        theta.expand(torch.Size([2, num_simulations, dim])),
+    )
+    assert estimator_dist_expanded.estimator == estimator_dist.estimator
+
+
+@pytest.mark.parametrize("num_dim", [3, 5])
+@pytest.mark.parametrize("num_trials", [1, 5])
+@pytest.mark.parametrize(
+    "trainer_cls, distribution_cls",
+    [
+        (NLE, ConditionalDensityEstimatorDistribution),
+        (NRE, RatioEstimatorDistribution),
+    ],
+)
+def test_pyro_gaussian_model(
+    trainer_cls,
+    distribution_cls,
+    num_dim,
+    num_trials,
+    num_simulations=500,
+    num_samples=500,
+    warmup_steps=500,
+):
+    """Test consistency of MCMC samples between the true and estimated likelihood."""
+    # Get data we will condition on
+    _, x_o = pyro_gaussian_model(num_trials=num_trials, num_dim=num_dim)
+
+    # Get MCMC samples from a model using the true likelihood
+    nuts_kernel = NUTS(pyro_gaussian_model)
+    mcmc = MCMC(nuts_kernel, num_samples=num_samples, warmup_steps=warmup_steps)
+    mcmc.run(x_o=x_o)
+    pyro_samples = torch.from_numpy(np.array(mcmc.get_samples()["theta"]))
+
+    # Simulated data from the true model and use it to train the estimator
+    prior = single_level_prior_theta(num_dim=num_dim)
+    theta = []
+    x = []
+    for _ in range(num_simulations):
+        ti, xi = pyro_gaussian_model(num_trials=1, num_dim=num_dim)
+        theta.append(torch.from_numpy(np.array(ti)))
+        x.append(torch.from_numpy(np.array(xi)))
+    theta = torch.stack(theta, dim=0).float()
+    x = torch.stack(x, dim=0).float().squeeze()
```

Comment:
I think I initially wrote this, but isn't there a vectorized way?

Reply:
Supposedly using [`Predictive`](https://docs.pyro.ai/en/stable/inference_algos.html?highlight=predictive#pyro.infer.predictive.Predictive) with `parallel=True` should work. I'll try it.

Reply:
That indeed works, and this is now moved to a fixture.